### ISSUE TITLE
plugin v1.1.5 produces crackling/staticy audio with OBS Studio 31.0.0-rc1

### ISSUE THREAD

**Issue by PaddyMac** [dimtpap/obs-pipewire-audio-capture#79]
The audio is fine with OBS Studio 30.2.3. It appears the plugin needs to be updated for the new OBS 31.0.0 after I compiled 31.0.0-rc1 and tested it.

**Comment by dimtpap** [comment]
I'm not having this issue.
Is the audio messed up the entire time?
Could you run OBS with `--verbose` and post here a small clip of a recording with this issue happening, along with the OBS log file?

**Comment by PaddyMac** [comment]
I need to do further testing. There may be multiple/confounding issues going on.

**Comment by PaddyMac** [comment]
Just following up. I have been testing various combinations of factors. At least SOME of the problem may just be Pipewire because I experience some static during gameplay from the game's music even if OBS isn't running, and that static seems to be caused by Firefox doing anything. So the game music static doesn't seem to be something I can pin on your plugin. I just need to do some more testing to figure out the cause of the distortion on my voice from the mic. I was also experiencing an issue with Intel Media driver which I filed a bug for, so there is also a confounding variable there. And so it might end up being that this bug was filed in error. But I'm leaving it open until I can rule out confounding factors.

**Comment by alterNERDtive** [comment]
If it’s related to games … some request unholy latencies in the sub millisecond range. Might be fixed by setting a reasonable one via environment variable for the game.

**Comment by PaddyMac** [comment]
Games and streaming/recording with OBS is what prompted me to file a bug and investigate this issue. And I suspected this plugin because the plugin was the cause of a similar issue several months ago. But it turned out that I was using an old version of the plugin when I upgraded OBS. Upgrading the plugin solved the issue. And the problem stayed gone for several months. So when the same or at least similar static/distortion cropped up again, I assumed it was the plugin again.

As it turned out, it seemed to be an issue largely stemming from system load because I am on an aging i7-3770s. The problem is not with this plugin. After a lot of frustration, the advice in this Reddit post https://www.reddit.com/r/linux_gaming/comments/1d1ef61/pipewire_audio_distortion_and_crackling_fix/ finally solved my problem. So I'd highly recommend anyone with static/crackling/distortion on Pipewire at least try this solution.
